src/quo2/components/calendar/calendar/years_list/view.cljs

```diff
@@ -0,0 +1,55 @@
+(ns quo2.components.calendar.calendar.years-list.view
+  (:require [react-native.core :as rn]
+            [quo2.theme :as theme]
+            [reagent.core :as reagent]
+            [quo2.foundations.colors :as colors]
+            [react-native.linear-gradient :as linear-gradient]
+            [quo2.components.calendar.calendar.utils :as utils]
+            [quo2.components.calendar.calendar-year.view :as calendar-year]
+            [quo2.components.calendar.calendar.years-list.style :as style]))
+
+(defn render-year
+  [year selected-year on-press]
+  [calendar-year/calendar-year
+   {:selected? (= year selected-year)
+    :on-press  #(on-press year)}
+   (str year)])
+
+(defn on-year-press
+  [on-change selected-year]
+  (do
+    #(reset! selected-year %)
+    (on-change selected-year)))
+
+(defn- separator
+  []
+  [rn/view {:style {:height 4}}])
+
+(defn- footer
+  []
+  [rn/view {:style {:height 32}}])
+
+(defn years-list-view-internal
+  [{:keys [on-change-year year theme]}]
+  (let [selected-year (reagent/atom year)]
+    [rn/view
+     {:style (style/container-years theme)}
+     [rn/flat-list
+      {:data                            (utils/generate-years (utils/current-year))
+       :key-fn                          str
+       :listKey                         :years-list
+       :inverted                        true
+       :shows-vertical-scroll-indicator false
+       :footer                          (footer)
+       :separator                       (separator)
+       :render-fn                       (fn [item]
+                                          (render-year item
+                                                       @selected-year
+                                                       #(on-year-press on-change-year item)))}]
+     [linear-gradient/linear-gradient
```

Comment:
Reagent was unnecessarily here, so i removed it

Reply:
> Reagent was unnecessarily here, so i removed it

Sorry I couldn't understand @mohsen-ghafouri, what do you mean by Reagent was unnecessary? My comment was related to the idea of optimizing the code by extracting `[linear-gradient/linear-gradient ...]` to a separate component in this file because it only depends on the `theme` value. It's just a suggestion though ;)

Reply:
sorry i misunderstood, resolved